### ISSUE TITLE
Getting an error

### ISSUE THREAD

**Issue by youngfreezy** [talal234/nodeChat#1]
Getting this error @talal234 

```
dyld: lazy symbol binding failed: Symbol not found: _node_module_register
  Referenced from: /Users/FareezAhmed/Desktop/nodeChat/node_modules/socket.io/node_modules/engine.io/node_modules/ws/build/Release/bufferutil.node
  Expected in: dynamic lookup

dyld: Symbol not found: _node_module_register
  Referenced from: /Users/FareezAhmed/Desktop/nodeChat/node_modules/socket.io/node_modules/engine.io/node_modules/ws/build/Release/bufferutil.node
  Expected in: dynamic lookup

Trace/BPT trap: 5
```

**Comment by tbalz2319** [comment]
Hmm I'll check it out

Regards,

Talal Balouch

On Tue, May 12, 2015 at 8:26 AM -0700, "youngfreezy" notifications@github.com wrote:

Getting this error boss:

dyld: lazy symbol binding failed: Symbol not found: _node_module_register
  Referenced from: /Users/FareezAhmed/Desktop/nodeChat/node_modules/socket.io/node_modules/engine.io/node_modules/ws/build/Release/bufferutil.node
  Expected in: dynamic lookup

dyld: Symbol not found: _node_module_register
  Referenced from: /Users/FareezAhmed/Desktop/nodeChat/node_modules/socket.io/node_modules/engine.io/node_modules/ws/build/Release/bufferutil.node
  Expected in: dynamic lookup

Trace/BPT trap: 5

**Comment by youngfreezy** [comment]
Dyld....Fais would have a field day with that

http://stackoverflow.com/questions/14630044/node-gyp-on-osx-10-7-5-dyld-lazy-symbol-binding-failed-symbol-not-found

**Comment by ddombrow** [comment]
I would try deleting your node_modules folder and running npm install again.  This should build any dependencies for your system.

Generally for a development project you don't want to check your node_modules folder in unless you're super paranoid about an update to one of your dependencies breaking your app.  Just add node_modules to .gitignore and delete the folder from the repo.  Anyone collaborating can just run npm install to get them since your dependencies are all listed in package.json.

**Comment by youngfreezy** [comment]
Interesting, yeah that worked thanks @ddombrow! Have a bunch of other errors I'm trying to work through now...

![image](https://cloud.githubusercontent.com/assets/11400324/7591369/8241588c-f89b-11e4-89de-3e8144441edf.png)

I think the first error is because jquery isn't loaded in the jade file...

**Comment by tbalz2319** [comment]
Fareez,

I think your Mac host-based firewall is on, you may want to disable it

or you can just open port 3700,

see the article below

http://www.macworld.com/article/1135888/firewalls.html

**Comment by youngfreezy** [comment]
I figured out the major source of confusion. in the server (index.js) file, the port was listed as 3500, and in the chat.js, it was listed as 3700 :8ball:

**Comment by youngfreezy** [comment]
also, jquery needed to be referenced first, that led to the '$ not defined' error going away:
http://stackoverflow.com/questions/2075337/uncaught-referenceerror-is-not-defined

**Comment by tbalz2319** [comment]
Sweet!

Regards,

Talal Balouch

On Tue, May 12, 2015 at 9:23 AM -0700, "youngfreezy" notifications@github.com wrote:

Closed #1.
